File: __Viejo/Formularios/Agregar/add_impuesto.vb
Public Class add_impuesto
    Private esRetencion As Boolean
    Private esPercepcion As Boolean
    Public id_impuesto As Integer
    Public Sub New()

        ' Esta llamada es exigida por el diseñador.
        InitializeComponent()

        ' Agregue cualquier inicialización después de la llamada a InitializeComponent().

    End Sub

    Public Sub New(ByVal _esRetencion As Boolean, ByVal _esPercepcion As Boolean)

        ' Esta llamada es exigida por el diseñador.
        InitializeComponent()

        ' Agregue cualquier inicialización después de la llamada a InitializeComponent().
        esRetencion = _esRetencion
        esPercepcion = _esPercepcion
    End Sub
    Private Sub cmd_ok_Click(sender As Object, e As EventArgs) Handles cmd_ok.Click
        If txt_nombre.Text = "" Then
            MsgBox("El campo 'Impuesto (%)' es obligatorio y está vacio")
            Exit Sub
        ElseIf txt_porcentaje.Text = "" Then
            MsgBox("El campo 'Porcentaje' es obligatorio y está vacio")
            Exit Sub
        End If

        Dim tmp As New impuesto

        tmp.nombre = txt_nombre.Text
        tmp.porcentaje = txt_porcentaje.Text
        tmp.esRetencion = chk_esRetencion.Checked
        tmp.esPercepcion = chk_esPercepcion.Checked
        tmp.activo = chk_activo.Checked

        If edicion = True Then
            tmp.id_impuesto = edita_impuesto.id_impuesto
            If updateImpuesto(tmp) = False Then
                MsgBox("Hubo un problema al actualizar el impuesto, consulte con su programdor", vbExclamation + vbOKOnly, "Centrex")
                closeandupdate(Me)
            End If
        Else
            id_impuesto = addImpuesto(tmp)
            If id_impuesto = -1 Then
                MsgBox("Hubo un problema al dar de alta el impuesto, consulte con su programdor", vbExclamation + vbOKOnly, "Centrex")
                closeandupdate(Me)
            End If
        End If

        If chk_secuencia.Checked = True Then
            txt_nombre.Text = ""
            txt_porcentaje.Text = ""
            chk_activo.Checked = True
        Else
            closeandupdate(Me)
        End If
    End Sub

    Private Sub cmd_exit_Click(sender As Object, e As EventArgs) Handles cmd_exit.Click
        closeandupdate(Me)
    End Sub

    Private Sub add_marcai_FormClosed(sender As Object, e As FormClosedEventArgs) Handles Me.FormClosed
        closeandupdate(Me)
    End Sub

    Private Sub add_descuento_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        chk_activo.Checked = True
        If edicion = True Or borrado = True Then
            chk_secuencia.Enabled = False
            txt_nombre.Text = edita_impuesto.nombre
            txt_porcentaje.Text = edita_impuesto.porcentaje
            chk_esPercepcion.Checked = edita_impuesto.esPercepcion
            chk_esRetencion.Checked = edita_impuesto.esRetencion
            chk_activo.Checked = edita_impuesto.activo
        End If

        If borrado = True Then
            txt_nombre.Enabled = False
            txt_porcentaje.Enabled = False
            chk_esRetencion.Enabled = False
            chk_esPercepcion.Enabled = False
            chk_activo.Enabled = False
            cmd_ok.Visible = False
            cmd_exit.Visible = False
            Me.Show()
            If MsgBox("¿Está seguro que desea borrar este impuesto?", vbYesNo + vbQuestion) = MsgBoxResult.Yes Then
                If (borrarImpuesto(edita_impuesto)) = False Then
                    If (MsgBox("Ocurrió un error al realizar el borrado del impuesto, ¿desea intectar desactivarlo para que no aparezca en la búsqueda?" _
                     , MsgBoxStyle.Question + MsgBoxStyle.YesNo)) = vbYes Then
                        'Realizo un borrado lógico
                        If updateImpuesto(edita_impuesto, True) = True Then
                            MsgBox("Se ha podido realizar un borrado lógico, pero el impuesto no se borró definitivamente." + Chr(13) +
                                "Esto posiblemente se deba a que el impuesto, tiene operaciones realizadas y por lo tanto no podrá borrarse", vbInformation)
                        Else
                            MsgBox("No se ha podido borrar el impuesto, consulte con el programador")
                        End If
                    End If
                End If
            End If
            closeandupdate(Me)
        End If

        If esRetencion Then
            chk_esRetencion.Checked = True
            chk_esRetencion.Enabled = False
            chk_esPercepcion.Enabled = False
            chk_secuencia.Enabled = False
            txt_porcentaje.Text = "0"
        ElseIf esPercepcion Then
            chk_esPercepcion.Checked = True
            chk_esPercepcion.Enabled = False
            chk_esRetencion.Enabled = False
            chk_secuencia.Enabled = False
            txt_porcentaje.Text = "0"
        End If
    End Sub
End Class

File: Formularios/Agregar/add_impuesto.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using Centrex.Models;

namespace Centrex
{
    public partial class add_impuesto
    {
   private bool esRetencion;
    private bool esPercepcion;
 public int id_impuesto;
        
        public add_impuesto()
        {
            // Esta llamada es exigida por el diseñador.
            InitializeComponent();
 }

        public add_impuesto(bool _esRetencion, bool _esPercepcion)
        {
            // Esta llamada es exigida por el diseñador.
      InitializeComponent();

      // Agregue cualquier inicialización después de la llamada a InitializeComponent().
    esRetencion = _esRetencion;
         esPercepcion = _esPercepcion;
     }

        private void cmd_ok_Click(object sender, EventArgs e)
        {
      if (string.IsNullOrEmpty(txt_nombre.Text))
      {
                Interaction.MsgBox("El campo 'Impuesto (%)' es obligatorio y está vacio");
                return;
            }
         else if (string.IsNullOrEmpty(txt_porcentaje.Text))
         {
      Interaction.MsgBox("El campo 'Porcentaje' es obligatorio y está vacio");
   return;
    }

    var tmp = new ImpuestoEntity();

            tmp.Nombre = txt_nombre.Text;
            tmp.Porcentaje = Conversions.ToDecimal(txt_porcentaje.Text);
    tmp.EsRetencion = chk_esRetencion.Checked;
    tmp.EsPercepcion = chk_esPercepcion.Checked;
            tmp.Activo = chk_activo.Checked;

            if (VariablesGlobales.edicion == true)
            {
   tmp.IdImpuesto = VariablesGlobales.edita_impuesto.IdImpuesto;
         if (impuestos.updateImpuesto(tmp) == false)
{
     Interaction.MsgBox("Hubo un problema al actualizar el impuesto, consulte con su programador", (MsgBoxStyle)((int)Constants.vbExclamation + (int)Constants.vbOKOnly), "Centrex");
       closeandupdate(this);
     }
     }
            else
 {
  id_impuesto = impuestos.addImpuesto(tmp);
   if (id_impuesto == -1)
                {
         Interaction.MsgBox("Hubo un problema al dar de alta el impuesto, consulte con su programador", (MsgBoxStyle)((int)Constants.vbExclamation + (int)Constants.vbOKOnly), "Centrex");
      closeandupdate(this);
 }
  }

            if (chk_secuencia.Checked == true)
 {
      txt_nombre.Text = "";
   txt_porcentaje.Text = "";
       chk_activo.Checked = true;
     }
      else
    {
  closeandupdate(this);
            }
        }

        private void cmd_exit_Click(object sender, EventArgs e)
        {
      closeandupdate(this);
        }

        private void add_marcai_FormClosed(object sender, FormClosedEventArgs e)
        {
            closeandupdate(this);
        }

        private void add_descuento_Load(object sender, EventArgs e)
        {
            chk_activo.Checked = true;
     if (VariablesGlobales.edicion == true | VariablesGlobales.borrado == true)
            {
           chk_secuencia.Enabled = false;
           txt_nombre.Text = VariablesGlobales.edita_impuesto.Nombre;
 txt_porcentaje.Text = VariablesGlobales.edita_impuesto.Porcentaje.ToString();
          chk_esPercepcion.Checked = VariablesGlobales.edita_impuesto.EsPercepcion ?? false;
          chk_esRetencion.Checked = VariablesGlobales.edita_impuesto.EsRetencion ?? false;
     chk_activo.Checked = VariablesGlobales.edita_impuesto.Activo;
            }

     if (VariablesGlobales.borrado == true)
            {
                txt_nombre.Enabled = false;
                txt_porcentaje.Enabled = false;
           chk_esRetencion.Enabled = false;
          chk_esPercepcion.Enabled = false;
 chk_activo.Enabled = false;
    cmd_ok.Visible = false;
         cmd_exit.Visible = false;
           Show();
      if (Interaction.MsgBox("¿Está seguro que desea borrar este impuesto?", (MsgBoxStyle)((int)Constants.vbYesNo + (int)Constants.vbQuestion)) == MsgBoxResult.Yes)
        {
        if (impuestos.borrarImpuesto(VariablesGlobales.edita_impuesto) == false)
    {
      if (Interaction.MsgBox("Ocurrió un error al realizar el borrado del impuesto, ¿desea intentar desactivarlo para que no aparezca en la búsqueda?", (MsgBoxStyle)((int)MsgBoxStyle.Question + (int)MsgBoxStyle.YesNo)) == Constants.vbYes)
   {
               // Realizo un borrado lógico
      if (impuestos.updateImpuesto(VariablesGlobales.edita_impuesto, true) == true)
  {
               Interaction.MsgBox("Se ha podido realizar un borrado lógico, pero el impuesto no se borró definitivamente." + "\r" + "Esto posiblemente se deba a que el impuesto tiene operaciones realizadas y por lo tanto no podrá borrarse", Constants.vbInformation);
   }
         else
           {
      Interaction.MsgBox("No se ha podido borrar el impuesto, consulte con el programador");
            }
      }
     }
          }
closeandupdate(this);
}

    if (esRetencion)
  {
         chk_esRetencion.Checked = true;
            chk_esRetencion.Enabled = false;
  chk_esPercepcion.Enabled = false;
     chk_secuencia.Enabled = false;
       txt_porcentaje.Text = "0";
  }
          else if (esPercepcion)
            {
                chk_esPercepcion.Checked = true;
  chk_esPercepcion.Enabled = false;
  chk_esRetencion.Enabled = false;
        chk_secuencia.Enabled = false;
          txt_porcentaje.Text = "0";
            }
        }
    }
}

